File: KSLC.js
const MAX_WEIGHT = 1020; 
const MAX_PEOPLE = 8; 
let currentWeight = 0; 
let peopleInLift = 0; 
let liftFull = false; 

function setup() {
    console.log("Lift Control System Initialized");
}

function loop() {
    currentWeight = readWeightSensor();
    peopleInLift = detectPeopleInLift();

    if (currentWeight > MAX_WEIGHT || peopleInLift >= MAX_PEOPLE) {
        liftFull = true;
        alertFullCapacity();
        skipFloors();
    } else {
        liftFull = false;
    }

    handleLiftMovements();
}

function readWeightSensor() {
    return Math.floor(Math.random() * (MAX_WEIGHT + 100));
}

function detectPeopleInLift() {
    return Math.floor(Math.random() * (MAX_PEOPLE + 1));
}

function alertFullCapacity() {
    console.log("Lift is full! Please wait.");
}

function skipFloors() {
    while (liftFull && peopleInLift < MAX_PEOPLE) {
        console.log("Skipping floors...");
        peopleInLift--;
    }
}

function handleLiftMovements() {
    if (!liftFull) {
        const buttonPressed = readButtonPress();
        if (buttonPressed) {
            moveLiftToFloor();
        }
    }
}

function readButtonPress() {
    return Math.random() < 0.5;
}

function moveLiftToFloor() {
    console.log("Moving lift to the desired floor...");
    setTimeout(() => {
        console.log("Lift has arrived at the floor.");
    }, 2000);
}

setup();
setInterval(loop, 1000);

File: KSLC.py
import time

current_weight = 0
max_weight = 1020  # Max weight in kg
people_in_lift = 0
max_people = 8
lift_full = False

def detect_people_in_lift():
    global people_in_lift
    if people_in_lift < max_people:
        people_in_lift += 1
    return people_in_lift

def check_weight():
    global current_weight
    return current_weight >= max_weight or people_in_lift >= max_people

def skip_floors():
    global people_in_lift
    while check_weight():
        print("Lift full, skipping floors...")
        time.sleep(2)  
        if people_in_lift > 0:
            people_in_lift -= 1  

def handle_lift_movements():
    if not lift_full:
        print("Lift moving to floor...")
        time.sleep(2)  
        print("Lift arrived at floor")

def main():
    global lift_full
    while True:
        people_in_lift = detect_people_in_lift()
        lift_full = check_weight()

        if lift_full:
            print("Lift is full!")
            skip_floors()
        else:
            handle_lift_movements()

        print(f"People in lift: {people_in_lift}, Current weight: {current_weight} kg")
        time.sleep(3) 

if __name__ == "__main__":
    main()
